src/makeCalculator.js

```diff
@@ -4,7 +4,45 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  let result = 0;
+
+  return {
+    add: function (num) {
+      result += num;
+
+      return this;
+    },
+    subtract: function (num) {
+      result -= num;
+
+      return this;
+    },
+    multiply: function (num) {
+      result *= num;
+
+      return this;
+    },
+    divide: function (num) {
+      if (num !== 0) {
+        result /= num;
+
+        return this;
+      } else {
+        throw new Error('Cannot divide by zero!');
+      }
+    },
+    reset: function (num) {
+      result = 0;
+
+      return this;
+    },
+    operate: function (operation, num) {
+      return operation.call(this, num);
+    },
```

Comment:
Use the method shorthand syntax for defining `operate`. Also, ensure that the `operation` parameter is a function before calling it to avoid runtime errors.